### PR TITLE
[FIX] Fix missing libraries in colab & Add Dynamically set the device

### PR DESCRIPTION
**Pull request by thliang01**
Add missing libraries in colab 

change **mps** to **cuda**

### CODE REVIEW

File: 1_instruction_tuning/notebooks/sft_finetuning_example.ipynb [hunk 1]
```diff
@@ -1,178 +1,215 @@
...
-        }
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "view-in-github",
+        "colab_type": "text"
+      },
+      "source": [
+        "<a href=\"https://colab.research.google.com/github/thliang01/smol-course/blob/main/1_instruction_tuning/notebooks/sft_finetuning_example.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
Comment (burtenshaw) [inline 1_instruction_tuning/notebooks/sft_finetuning_example.ipynb:10]:
```suggestion
        "<a href=\"https://colab.research.google.com/github/huggingface/smol-course/blob/main/1_instruction_tuning/notebooks/sft_finetuning_example.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```

**Review by burtenshaw — COMMENTED** [review]
Thanks. This looks great. I just want to check that the colab links point to the main notebooks.

**Comment by thliang01** [comment]
@burtenshaw I think the notebook should start in empty

**Comment by burtenshaw** [comment]
> @burtenshaw I think the notebook should start in empty
That's cool. Some people like to read the notebooks in the browser, and things like the dataset viewer output are quite informative for that. But it's up to you!

Merge the changes from `main` and LGTM!

**Pull request merged**